client/src/hooks/reducers/feedReducers.js

```diff
@@ -51,6 +52,14 @@ export const postsReducer = (state = postsState, action) => {
       return { ...state, status: SET_POSTS, posts: action.posts };
     case ERROR_POSTS:
       return { ...state, status: ERROR_POSTS, posts: [] };
+    case SET_LIKE:
+      return { ...state,
+        posts: state.posts.map((post) => (
```

Comment:
I think there's a more efficient to update the liked state than to loop through every post. Perhaps we could create a reducer that handles actions for a single post?

On the same note, should we create a separate `postReducer` to handle actions like this? In the future we need to handle things like commenting, liking comments, error handling, etc. So I think adding all the state logic here for a single post may be bloating the code and logic.

Reply:
adding logic in reducers makes it easy to test input / output from functions (like redux docs recommend as well)
post reducer sounds nice... i'm a bit confused at the moment how we are using redux and hooks with useReducer, maybe we should get rid of redux if we are only using it for the authentication & user ... 
to make it more efficient, maybe the state should be normalized and changed to have `posts: { id: value }` as an object like a map with keys and values

Reply:
Will the backend be changed to reflect `id: values` or do I need to do it in the frontend?
If we use a postReducer, how it will be filled? Do all results have a reducer or It will be filled when the user interacts with a post? Which information will store postsRedurer and which postReducer? If postReducer is updated, it needs to update postsReducer post' information too?

Reply:
The common best practice to normalize the state is to change arrays with objects into an object with key/value and keep a separate list with only the id's in memory and also use that in other places where you need to present the data in a different way... there is extensive documentation about this on the redux docs https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape and also some related blog posts ... I think this is probably a good idea to also do this with the useReducer hooks

Reply:
````
? Currently, the posts service doesn't return if a post was liked by a user, it's missing it, how do we handle this?
````
Yes good question, I also thought about that when looking at the data model, this was also my idea but since for our use case posts are mostly "read-heavy" the data model is designed for that, and I think we can get that info from the API, maybe check with @Naraujo13  because there might be a change needed, for example if authenticated user requests the list of posts, add an extra boolean if their user id is in the "likes" list

Reply:
> The common best practice to normalize the state is to change arrays with objects into an object with key/value and keep a separate list with only the id's in memory and also use that in other places where you need to present the data in a different way... 

Ok. I will do the change in SET_POSTS dispatch after receive the service response.

Reply:
@robinv85 I can add that feature at the `/posts` endpoint.
Just adding a boolean that checks if the user liked the post or not should be easy. 

@silenceway could you open an issue specifying these changes? Also, review if you need any more additional info from the posts' backend.

Reply:
yeah ok wasn't sure if this was already implemented, seems like we forgot about that, but this is also a good reason why we should use the authenticated hook in the API for this request to get the user id, but then optional...
I was actually wondering if this was also allowed by the OpenAPI standard because I haven't seen it yet, and not sure if this will show in our auto-generated API documentation, but seems like it should be possible https://github.com/OAI/OpenAPI-Specification/issues/14#issuecomment-297457320

Reply:
Updated the reducer.